decisions/utils.go

```diff
@@ -82,14 +84,18 @@ func (node *Node) getAllSubNodes(includeActions, includeConditions bool) []*Node
 	return nodes
 }
 
-// MutateNode randomly mutates a single node of a tree
+// MutateNode randomly mutates a single node of a tree. This function
+// should only be called on root tree nodes because it uses the tree size.
 func MutateNode(node *Node) {
 	// pick a random node anywhere in the decision tree
 	allSubNodes := node.getAllSubNodes(true, true)
 	toMutate := allSubNodes[rand.Intn(len(allSubNodes))]
 
+	treeSize := node.Size()
+	maxTreeSize := config.MaxDecisionTreeSize()
+
 	if toMutate.IsAction() {
-		if rand.Intn(2) == 0 {
+		if rand.Intn(2) == 0 && treeSize < maxTreeSize {
```

Comment:
Might need to make this treeSize  < (maxTreeSize - 1), since this mutation adds two nodes to the tree

Reply:
Oh that's a good point, originally I was thinking of it in terms of tree height.

Reply:
Ah, that makes sense